src/github/client/github-app-client.ts

```diff
@@ -47,8 +47,8 @@ export class GitHubAppClient extends GitHubClient {
 			handleFailedRequest(this.logger)
 		);
 		this.axios.interceptors.response.use(
-			instrumentRequest(metricHttpRequest.github),
-			instrumentFailedRequest(metricHttpRequest.github)
+			instrumentRequest(metricHttpRequest.github, this.restApiUrl),
```

Comment:
Shall we move the `this.restApiUrl` to above, and only passing generic tags in the `instrucmentRequest`? Like


```javascript
const gitHubVersion = isCloudOrServerHost(url);
const extraTags = { gitHubVersion};
this.axios.interceptors.response.use(
			instrumentRequest(metricHttpRequest.github, extraTags),
			instrumentFailedRequest(metricHttpRequest.github, extraTags)
		);
```

Reply:
"What's the benefit of..."
     <--- I am thinking from `instrumentRequest` point of view, what does it really need from consumer? Does it need the full url? Or does it just need a tag provided by consumer? And what if there's future need to add more tags from consumer? 

"Also, the metrics handle the value for gitHubVersion a little differently..."
    <--- I would say something like ` { client: "axios", status: status, ...extraTags }` should do the trick in both cases. 

--------------

But having say that, it seems you have thought about the both cases thoroughly, so I am happy to follow your idea, and close this thread. Please feel free to close this thread.